api/src/legal-documents/infrastructure/repositories/legal-document.repository.js

```diff
@@ -21,4 +21,22 @@ const findLastVersionByTypeAndService = async ({ type, service }) => {
   return new LegalDocument(documentVersionDto);
 };
 
-export { findLastVersionByTypeAndService };
+/**
+ * Creates a new legal document in the database.
+ *
+ * @param {Object} params - The parameters.
+ * @param {string} params.type - The type of the legal document.
+ * @param {string} params.service - The service associated with the legal document.
+ * @param {Date} params.versionAt - The date of the legal document version.
+ * @returns {Promise<LegalDocument>} The newly created legal document.
+ */
+const createLegalDocument = async ({ type, service, versionAt }) => {
```

Comment:
Tu peux juste le nommé "create" car on est déjà dans le repo `legal-document.repository`
```suggestion
const create = async ({ type, service, versionAt }) => {
```